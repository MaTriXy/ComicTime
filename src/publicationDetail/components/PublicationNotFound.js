import React from "react";
import { translateComponent } from "../../i18n/i18n";
import ComicTimeLogo from "../../baseComponents/comicTimeLogo/ComicTimeLogo";

const style = {
  container: {
    width: "100%",
    height: "100%"
  },
  iconContainer: {
    position: "absolute",
    height: "200px",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)"
  }
};

class PublicationNotFound extends React.Component {
  render() {
    const { t } = this.props;
    return (
      <div style={style.container}>
        <div style={style.iconContainer}>
          <ComicTimeLogo />
          <p>
            {t("publicationNotFoundMessage")}{" "}
            <span role="img" aria-label="Sad">
              &#128546;
            </span>
          </p>
        </div>
      </div>
    );
  }
}

export const PurePublicationNotFound = PublicationNotFound;
export default translateComponent(PurePublicationNotFound);
