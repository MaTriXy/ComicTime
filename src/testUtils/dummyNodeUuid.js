jest.mock("node-uuid", () => {
  return {
    v4() {
      return "uuid";
    }
  };
});
