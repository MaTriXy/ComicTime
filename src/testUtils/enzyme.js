import React from "react";
import Enzyme from "enzyme";
import { createSerializer } from "enzyme-to-json";
import toJson from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import { MuiThemeProvider } from "material-ui";
import comicTimeTheme from "../theme";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));
Enzyme.configure({ adapter: new Adapter() });

export function mockHistory() {
  return {
    go: jest.fn(),
    goBack: jest.fn()
  };
}

export function translator(key) {
  return key;
}

export function mountWithMuyThemeProvider(component) {
  return Enzyme.mount(
    <MuiThemeProvider muiTheme={comicTimeTheme}>{component}</MuiThemeProvider>
  );
}

export function mountComponent(component) {
  return Enzyme.mount(component);
}

export function mountComponentAsJson(component) {
  return toJson(mountComponent(component));
}

export function shallowComponent(component) {
  return Enzyme.shallow(component);
}

export function shallowComponentAsJson(component) {
  return toJson(shallowComponent(component));
}

export function json(component) {
  return toJson(component);
}
