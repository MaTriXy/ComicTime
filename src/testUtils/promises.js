export async function flushPromises() {
  return new Promise(resolve => setImmediate(resolve));
}
