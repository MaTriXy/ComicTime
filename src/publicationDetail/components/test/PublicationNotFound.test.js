jest.mock("../../../i18n/i18n");
jest.mock(
  "../../../baseComponents/comicTimeLogo/ComicTimeLogo",
  () => "ComicTimeLogo"
);

import React from "react";
import { shallowComponentAsJson, translator } from "../../../testUtils/enzyme";
import { PurePublicationNotFound } from "../PublicationNotFound";

describe("PublicationNotFound", () => {
  it("shows the applicaiton logo and a message to indicate the publication couldn't be loaded", () => {
    const component = shallowComponentAsJson(
      <PurePublicationNotFound t={translator} />
    );
    expect(component).toMatchSnapshot();
  });
});
