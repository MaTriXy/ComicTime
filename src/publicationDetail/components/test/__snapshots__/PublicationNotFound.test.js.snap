// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationNotFound shows the applicaiton logo and a message to indicate the publication couldn't be loaded 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "height": "200px",
        "left": "50%",
        "position": "absolute",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
      }
    }
  >
    <ComicTimeLogo />
    <p>
      publicationNotFoundMessage
       
      <span
        aria-label="Sad"
        role="img"
      >
        ðŸ˜¢
      </span>
    </p>
  </div>
</div>
`;
